src/testnet/mem_pool.rs

```diff
@@ -0,0 +1,84 @@
+use std::thread;
+use std::time;
+use std::fs;
+use std::io::Read;
+use std::io::BufReader;
+use std::io::prelude::*;
+
+use burnchains::{Txid};
+use chainstate::stacks::{StacksTransaction};
+use net::StacksMessageCodec;
+
+pub trait MemPool {
+    fn poll(&mut self) -> Vec<StacksTransaction>;
+    fn start(&mut self);
+    fn stop(&mut self);
+    fn handle_incoming_tx(&mut self, tx: StacksTransaction);
+    fn archive_tx(&mut self, tx: StacksTransaction);
+}
+
+#[derive(Clone)]
+pub struct MemPoolFS {
+    path: String,
+}
+
+impl MemPoolFS {
+    pub fn new(path: &str) -> Self {
+        match fs::create_dir_all(path) {
+            Ok(_) => {},
+            Err(_) => panic!("Error while creating dir at path {}", path)
+        };
+
+        Self {
+            path: path.to_string(),
+        }
+    }
+}
+
+impl MemPool for MemPoolFS {
+
+    fn poll(&mut self) -> Vec<StacksTransaction> {
+        let txs_paths = fs::read_dir(&self.path).unwrap();
+        let mut decoded_txs = vec![];
+
+        for tx in txs_paths {
+            if let Ok(tx) = tx {
+                let path = tx.path();
+                if path.is_dir() {
+                    continue;
+                }
+
+                let file = fs::File::open(path.clone()).unwrap();
+                let mut reader = BufReader::new(file);
+                assert!(reader.buffer().is_empty());
+                reader.fill_buf().unwrap();
+                let encoded_tx: Vec<u8> = reader.buffer().to_vec();
+                let mut index = 0;
+                let tx = StacksTransaction::deserialize(&encoded_tx, &mut index, encoded_tx.len() as u32).map_err(|_e| {
+                    eprintln!("Failed to decode transaction {:?}", _e);
+                    panic!();
+                }).unwrap();
```

Comment:
Instead of panicking, can we just remove this transaction and continue processing the directory?  Eventually, we'll be running a public version of this where people can submit arbitrary transactions (and we don't want them to crash the testnet).

Reply:
Addressed with https://github.com/blockstack/blockstack-core/pull/1202/commits/fd1fda5fe76081cc633868fa6237fb963af6c1da